src/permissionSelectors.js

```diff
@@ -40,6 +42,105 @@ export function roleIsAtLeast(thisRole: RoleT, thresholdRole: RoleT): boolean {
   return thisRole <= thresholdRole; // Roles with more privilege have lower numbers.
 }
 
+/**
+ * Whether the user has passed the realm's waiting period to be a full member.
+ *
+ * See:
+ *   https://zulip.com/api/roles-and-permissions#determining-if-a-user-is-a-full-member
+ *
+ * To determine if a user is a full member, callers must also check that the
+ * user's role is at least Role.Member.
+ *
+ * Note: If used with useSelector, the value will be out-of-date, though
+ * realistically only by seconds or minutes at most; see implementation.
+ */
+export function getHasUserPassedWaitingPeriod(state: PerAccountState, userId: UserId): boolean {
+  const { waitingPeriodThreshold } = getRealm(state);
+  const { date_joined } = getUserForId(state, userId);
+
+  return (
+    differenceInDays(
```

Comment:
I tend to get nervous when I see a function like this that there are several different things such a name could mean, and it's not clear whether what the function actually does is a thing we want.

In this case, looking at the implementation in `node_modules/date-fns/differenceInDays/index.js`, there is a nice jsdoc:
```js
/**
 * @name differenceInDays
 * @category Day Helpers
 * @summary Get the number of full days between the given dates.
 *
 * @description
 * Get the number of full day periods between two dates. Fractional days are
 * truncated towards zero.
 *
 * One "full day" is the distance between a local time in one day to the same
 * local time on the next or previous day. A full day can sometimes be less than
 * or more than 24 hours if a daylight savings change happens between two dates.
 *
 * To ignore DST and only measure exact 24-hour periods, use this instead:
 * `Math.floor(differenceInHours(dateLeft, dateRight)/24)|0`.
```

and it does make clear what choices it's made about the edge cases -- and it's not something we want, because it will give different answers in different timezones.

The thing we actually want is also simple to compute, fortunately:
* Subtract the one Date object from the other: `new Date() - new Date(date_joined)`. (Or `Date.now() - Date.parse(date_joined)`, which is equivalent but I think a bit more explicit.) This gives the length of time between them in milliseconds.
* Divide by the number of milliseconds in a day, which is `86400_000`. (Aka 24 * 60 * 60 * 1000.)
* That's the length of the interval in days, as a floating-point number. Compare it by `>= waitingPeriodThreshold`.